### DISCUSSION TITLE
data hazard in pipelining

### DISCUSSION THREAD

**Discussion by Anasa1** [mortbopet/Ripes#97] [Q&A] [answered]
hello... Is it possible to run and detect data hazard pipelined in this simulator? Is there any tutorial or example that I can follow as I'm still a beginner with this simulator?

**Comment by mortbopet** [comment, marked as answer]
Hi @Anasa1
I'd recommend you to take a look at introductory books to computer architecture, such as Computer Organization and Design The Hardware/Software Interface: RISC-V Edition.
Apart from that, the [Ripes introduction](https://github.com/mortbopet/Ripes/wiki/Ripes-Introduction) also shows a data hazard, and how this is visualized in the pipeline.
